File: src/main/kotlin/com/ks/culinario/domain/model/ShoppingList.kt
package com.ks.culinario.domain.model

import kotlinx.serialization.Serializable

data class ShoppingList(
    val id: Int,
    val title: String,
    val description: String,
    val items: List<ShoppingListItem>
) {
    @Serializable
    data class ShoppingListItem(val name: String, val amount: Int)
}

File: dotnet/Culinario.Domain/Models/ShoppingList.cs
namespace Culinario.Domain.Models;

public class ShoppingList
{
    public int Id { get; set; }
    public string Title { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    public List<ShoppingListItem> Items { get; set; } = new();
}

public class ShoppingListItem
{
    public string Name { get; set; } = string.Empty;
    public int Amount { get; set; }
}
